src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -279,9 +295,9 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
             isAsleep = compound.getBoolean(TAG_ASLEEP);
         }
 
-        if (level > MAX_CITIZEN_LEVEL)
+        if (job != null && levels.getOrDefault(job.getName(), 0) > MAX_CITIZEN_LEVEL)
```

Comment:
shouldnt this check be in the for loop reading the value? To check the Max of every value, not just the current jobs

Reply:
This is actually only a hotfix to reset workers which had a too high level before we added the level cap. That's something we can delete in the future again.

Reply:
well if we were to ever change the constant again you'd need it :D always good to validate the data upon rereading^^